source/extensions/filters/http/cdn/cdn_loop_utils.cc

```diff
@@ -0,0 +1,34 @@
+#include "extensions/filters/http/cdn/cdn_loop_utils.h"
+
+#include <algorithm>
+
+#include "common/common/statusor.h"
+
+#include "extensions/filters/http/cdn/cdn_loop_parser.h"
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cdn {
+
+StatusOr<int> countCdnLoopOccurrences(absl::string_view header, absl::string_view cdn_id) {
```

Comment:
Glancing at the spec, I think we'd want to handle HTTP/1.1. style multi-entry headers, where you can have
Foo: bar
Foo: eep
not just 
Foo: bar,eep
Are you assuming any such headers would be appended together or should this take 
vector<absl::string_view> header_values ?

Reply:
I'm assuming that any such headers would be appended together.  I'm targeting what I get out of `Envoy::Http::HeaderMap`, which returns a `HeaderEntry`, which returns a single `HeaderString&` as its value.